component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/mepin/MepinTransactions.java

```diff
@@ -58,24 +60,30 @@
      * @param expiryTime         the expiry time
      * @return the transaction response
      * @throws IOException
+     * @throws AuthenticationFailedException
      */
-    protected String createTransaction(String mepinID, String sessionID, String url,
-                                       String username, String password, String clientId,
+    protected String createTransaction(String mepinID, String url,
+                                       String username, String password, String appId,
                                        String header, String message, String shortMessage,
                                        String confirmationPolicy, String callbackUrl,
                                        String expiryTime) throws IOException, AuthenticationFailedException {
         if (log.isDebugEnabled()) {
             log.debug("Started handling transaction creation");
         }
-        String query = String.format(MepinConstants.MEPIN_QUERY, URLEncoder.encode(sessionID, MepinConstants.CHARSET),
-                URLEncoder.encode(shortMessage, MepinConstants.CHARSET),
-                URLEncoder.encode(header, MepinConstants.CHARSET), URLEncoder.encode(message, MepinConstants.CHARSET),
-                URLEncoder.encode(clientId, MepinConstants.CHARSET),
-                URLEncoder.encode(mepinID, MepinConstants.CHARSET), URLEncoder.encode(expiryTime, MepinConstants.CHARSET),
-                URLEncoder.encode(callbackUrl, MepinConstants.CHARSET),
-                URLEncoder.encode(confirmationPolicy, MepinConstants.CHARSET)
-        );
-        String response = postRequest(url, query, username, password);
+        MepinTransaction mepinTransaction = new MepinTransaction();
+        mepinTransaction.setAction(MepinConstants.TRANSACTIONS_CREATE);
+        mepinTransaction.setApp_id(appId);
+        //set as a long random hex string, 32 bytes.
+        mepinTransaction.setIdentifier(getRandomHexString());

```

Comment:
Cann't we use UUID generate to generate Identifier

Reply:
in the spec, they have mentioned using a random hex string with 32 bytes long. but the UUID would be 36 bytes long with the hyphens.

Reply:
But I tested with replacing the logic with generated UUID from java and there was no issue. hence replacing the current logic